### PR TITLE
Making status take into account the PR states

### PR DESCRIPTION
**Pull request by gauntface**
This enables auto-merge UI with new repo that includes status for each PR and sets the correct status type for outgoing PR's

### CODE REVIEW

**Comment by gauntface** [comment]
@samuelli I'm really struggling to make the pixel tests pass one after the other (even locally).

My hunch is that the test server is using the same name for the tests but the results are no difference because of the paging.

File: src/server/apis/dash-data/fetch-outgoing-data.ts [hunk 1]
```diff
@@ -153,9 +153,27 @@ async function getAllPRInfo(
         });
       }
 
+      let latestCommit: commitFieldsFragment|null = null;
```
Comment (samuelli) [inline src/server/apis/dash-data/fetch-outgoing-data.ts]: Move this line down to where its used. L165.

File: src/server/apis/dash-data/fetch-outgoing-data.ts [hunk 2]
```diff
@@ -260,7 +280,17 @@ function getStatus(
...
+        const state = latestCommit.status.state;
+        if (state === 'PENDING') {
+          outgoingStatus = {type: 'StatusChecksPending'};
+        } else if (state === 'ERROR' || state === 'FAILURE') {
+          outgoingStatus = {type: 'StatusChecksFailed'};
+        }
+      }
+      if (outgoingStatus.type === 'UnknownStatus') {
+        outgoingStatus = {type: 'PendingMerge'};
+      }
```
Comment (samuelli) [inline src/server/apis/dash-data/fetch-outgoing-data.ts]: What about when status checks are passing?

Comment (gauntface) [reply src/server/apis/dash-data/fetch-outgoing-data.ts]: the two remaining states are no states and success states, both should result in PendingMerge for the dashboard state.

**Comment by samuelli** [comment]
> @samuelli I'm really struggling to make the pixel tests pass one after the other (even locally).
> 
> My hunch is that the test server is using the same name for the tests but the results are no difference because of the paging.

Just leave these then.

**Pull request merged**